pkg/cluster/meta.go

```diff
@@ -0,0 +1,55 @@
+package cluster
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MetaOptions allows to add additional settings for the object being created through a chain
+// of functions which are applied on metav1.Object before actual resource creation.
+type MetaOptions func(obj metav1.Object) error
+
+func ApplyMetaOptions(obj metav1.Object, opts ...MetaOptions) error {
+	for _, opt := range opts {
+		if err := opt(obj); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func WithOwnerReference(ownerReferences ...metav1.OwnerReference) MetaOptions {
+	return func(obj metav1.Object) error {
+		obj.SetOwnerReferences(ownerReferences)
+		return nil
+	}
+}
+
+func WithLabels(labels ...string) MetaOptions {
```

Comment:
In some other package probably, but maybe a higher level version as well for unified labeling; WithUnifedLables(dsci, dsc, component)?